### PR TITLE
[SPARK-21298][spark core]Don't need to modified name when it's added the same ids

### PR DESCRIPTION
**Pull request by Geek-He**
## What changes were proposed in this pull request?
It's not need to replace name when the key has exsited.
I have a stage,named freedom and id is 0,  then adding a stage that named nofreedom and id is 0. In this case, i think it's not need to modified the stage name. returns existing values only. 

## How was this patch tested?
Unit test

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info).name = info.name }
+    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info)}
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala]: It's not clear this is valid. You are updating the whole info object, not just the name, in the case that the ID existed. Why is this needed?

Comment (Geek-He) [inline core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala]: In the case  that the ID existed, the other member of the  whole info object is not updated, so why do you want to update the member name individually?

Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala]: That's the question to you -- why _don't_  you just want to do that?. You're actually reverting the change in SPARK-9044 (use `git blame` to understand who changed this and when.).

**Comment by jiangxb1987** [comment]
Since this is not a bug, could you please close this? @Geek-He Thank you!

**Comment by Geek-He** [comment]
ok
